### ISSUE TITLE
re-releasing 1807 (how to?)

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#128]
1807 has a bug where you can't retreat in naval combats. @ron-murhammer has a PR with a fix in, but there is an open question of how to get the bug fix out to users.

There are four options I see us having:
1. Apply bug fix to 1807 release branch and rebuild artifacts. Notify users to re-download and install again.
2. Apply bug fix to 1807 release branch  and release as 1809
3. Apply bug fix to current master and release as 1809 (release wit the  "old ant" release process)
4. Apply bug fix to current master along with Travis CI changes. We then do normal "new gradle" release process, doing a full regression test, testing the new CI process at the same time.

**Comment by DanVanAtta** [comment]
We should probably communicate this bug in the lobby. It'll let players know to work around until we get a patch released.

**Comment by gaborbernat** [comment]
this is exactly why I proposed adding an extra number at the end to allow for bug releases. It should be like 1.8.0.7.0, 1.8.0.7.1, etc... but perhaps we can skip the third 0, and make it 1.8.7.0 and so on

**Comment by DanVanAtta** [comment]
My initial response I copy pasted into a new issue, #129 for release numbers, it's a juicy topic! We do have a pretty serious bug that we need to focus on getting fixed and patched. Pretty clearly the version number probably needs to be updated. So the next version would likely be 1809 (rather than 1810 or 1808). If  that is case, then we are left with just two options:  3 and 4.

**Comment by veqryn** [comment]
I think 1809 would be best, since it is really hard to notify people that they need to re-download, unless we bump the number.

**Comment by ron-murhammer** [comment]
Agree. I think we should target 1809. The only question is what changes we
should release. I think we can go for master plus the fix if you can look
over things. I also put out a pre-release with that.

**Comment by DanVanAtta** [comment]
:+1: , @ron-murhammer , would you mind pushing into github as well?

**Comment by DanVanAtta** [comment]
PR submitted to update 1.8.0.9 change notes: https://github.com/triplea-game/triplea/pull/130

**Comment by veqryn** [comment]
I will need to review the scroll speed thing first.
I actually hate the scroll zone, so I am not sure I like that patch.....

**Comment by DanVanAtta** [comment]
It's already been merged to master. "Revert" in git can have unintended consequences, it is not quite as similar to SVN. Subjective opinion of a feature that has been improved I don't see as being relevant. I would welcome creation of an option to turn the feature off altogether. In the meantime the feature is improved. Why reject work that was done to make something better, it does not make much sense, particularly when the fallback is to go back to the less useful scroll zones that were generating complaints for being small and scrolling too slowly.

Considering it has been reviewed, tested, and now regression tested, I'm not sure what you would review the feature for. TBH it sounds like you woudl be wasting your time, spend it instead on removing the feature and submitting a PR if you think that is the right way to go.

**Comment by ron-murhammer** [comment]
I merged the changes. I recommend you release 1809 from master.

**Comment by ron-murhammer** [comment]
So it appears our 2 options are:
1. Apply bug fix to current master and release as 1809 (release with the "old ant" release process)
2. Apply bug fixes to 1807 release branch and release as 1809

I think we should try to do the first one and have the second as our backup plan if we find issues. I've already uploaded a pre-release with everything that's in master. I think we get a few people to test things out in the community and go through the release regression testing.

@DanVanAtta @gaborbernat do you think you guys can run through the release testing: https://github.com/triplea-game/triplea/wiki/Release-Testing

**Comment by gaborbernat** [comment]
oh, that list is really long and troublesome, unfortunately the most I can do today is build and test the all platform local game on Linux and Windows; if that helps... :cry:

**Comment by DanVanAtta** [comment]
Why not venture into the new gradle process? Seems like regression testing with the new release would be the most efficient use of our limited time. Same with going with option number 1, regression testing one feature at a time is a waste of time and effort.

@ron-murhammer , build and deploy the artifacts, before then they cannot be properly tested. Notably, if I build, or if @gaborbernat , then who knows what we will have tested in relation to the actual build artifact.

Secondly, please upload it to github. 3 minute DL from github is nice, the context switching waiting for a 25 minute DL from sourceforge unnecessarily is a waste of time.

I can probably do some more testing tonight, please notify when the artifacts have been built and uploaded.

**Comment by gaborbernat** [comment]
I just finished (at least think so adding the new installer, and
gradle only zip releases). However it's not yet merged in, currently
it's in mine clone, installer branch, so would probably require a
longer testing process; before it would be ready for user release. :)

On 8/30/15, Dan Van Atta <notifications@github.com> wrote:
> Why not venture into the new gradle process? Seems like regression testing
> with the new release would be the most efficient use of our limited time.
> Same with going with option number 1, regression testing one feature at a
> time is a waste of time and effort.
>
> @ron-murhammer , build and deploy the artifacts, before then they cannot be
> properly tested. Notably, if I build, or if @gaborbernat , then who knows
> what we will have tested in relation to the actual build artifact.
>
> Secondly, please upload it to github. 3 minute DL from github is nice, the
> context switching waiting for a 25 minute DL from sourceforge unnecessarily
> is a waste of time.
>
> I can probably do some more testing tonight, please notify when the
> artifacts have been built and uploaded.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/triplea-game/triplea/issues/128#issuecomment-136162359


-- 
 Bernát Gábor
Student -  Budapest University of Technology and Economics - Computer
Engineering, M.Sc. - Budapest, Hungary
System Integrator - Gravity R&D | Rock Solid Recommendations - Budapest
office | 5-7 Expo ter | H-1101 | Budapest | Hungary

**Comment by DanVanAtta** [comment]
A slightly longer testing process now saves a lot of time in the long run. It also lets us get the Mac DMG built sooner - btw, if we can get that built in the next 8 hours, I can test the mac release (but I can't change java on that system, so it needs to be the newer DMG that can run in Java7 or Java8. I still think the Java6 requirement of the old Mac DMG is effectively a P0 bug for that artifact now. Either we should stop building the Java6 only Mac DMG or update it).

**Comment by gaborbernat** [comment]
I've just rebased my installer on master, so I'll create on my machine a DMG file. Event if we're not going to use it, for merging in the new installer trying it out will be really beneficial :)

**Comment by gaborbernat** [comment]
I've uploaded it to https://goo.gl/e1MS3v, both the MAC and Windows installers have Java installer backed in (actually it uses the backed in Java no matter what). It's noted as 1.8.0.7 as I've not bumped the version number in the builder, however it's 1.8.0.9 trust me. Any feedback is well appreciated. Still not sure if we should use the new build as this would delay the release I'd say a couple of days.

**Comment by DanVanAtta** [comment]
Can you elaborate why a few days? Do you expect there to be a lot of problems with the new installers?

As is, we need to install on linux, mac and windows, play a game round, and then do some other testing like launch a PBEM game. Changing how the installer is made, I don't quite see adding that much effort to that process. Perhaps I am overlooking something.

**Comment by gaborbernat** [comment]
Ideally, it should just work. But experience tells me there will be
some bugs :S nothing special in mind, just experience :)

On 8/30/15, Dan Van Atta <notifications@github.com> wrote:
> Can you elaborate why a few days? Do you expect there to be a lot of
> problems with the new installers?
>
> As is, we need to install on linux, mac and windows, play a game round, and
> then do some other testing like launch a PBEM game. Changing how the
> installer is made, I don't quite see adding that much effort to that
> process. Perhaps I am overlooking something.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/triplea-game/triplea/issues/128#issuecomment-136170270


-- 
 Bernát Gábor
Student -  Budapest University of Technology and Economics - Computer
Engineering, M.Sc. - Budapest, Hungary
System Integrator - Gravity R&D | Rock Solid Recommendations - Budapest
office | 5-7 Expo ter | H-1101 | Budapest | Hungary

**Comment by gaborbernat** [comment]
try out the installers and we'll see :) I've tried at least at install level each of them

**Comment by DanVanAtta** [comment]
I'm trying the Mac installer now. I managed to kill my windows system recently, so cannot test there for a few more days yet.

**Comment by DanVanAtta** [comment]
Mac DMG installer has problems.

Application installer can't be launched due to this error:

https://mail.google.com/mail/u/0/?ui=2&ik=dbad076eab&view=fimg&th=14f7fe30e4e467a0&attid=0.1&disp=emb&realattid=ii_14f7fe2fd5884f93&attbid=ANGjdJ_cH0h4VAHTQn8NQ_KQznaT4D22tGxKY4WAZR2pYiy7KFsNX3IM9ULwuXgcH99C1_2p2SB-XrB4VnBKUzHdK6kC90xvbnkDmY_DONSg2Sxr1x-sakumTqxgjGo&sz=w834-h490&ats=1440959660221&rm=14f7fe30e4e467a0&zw&atsh=1

Secondly, looks like the packaging is off, the installer is not executable or something. Typically when a DMG file is opened you see an icon for the application, and an Applications folder icon as well. To install you drag and drop the app to the folder. In this DMG, there was no folder. Dragging the app to Applications folder copied the installer. Then when trying to launch the installer again, get the same message.

**Comment by DanVanAtta** [comment]
On the plus side, a Java7 or Java8 Mac DMG is better than feature parity. java6 Mac only DMG is a broken artifact, requiring Java6 is like requiring windows XP. So even if we can get the windows installer and all_platforms.zip out, if it takes a few days to get the Mac DMG done, not a bid deal IMO.

**Comment by gaborbernat** [comment]
Can't  see your link :)

**Comment by DanVanAtta** [comment]
Just updated now, pls refresh

**Comment by DanVanAtta** [comment]
Heh.. reading through the settings, looks like there is a preference somewhere i need to update?

**Comment by DanVanAtta** [comment]
Gonna try to find that preference and will retry.

**Comment by gaborbernat** [comment]
oh, see please MacOS_users_read_this_first.txt in the root of tripleA

**Comment by gaborbernat** [comment]
that's a known limitation of the old installer too :)

**Comment by gaborbernat** [comment]
maybe to bypass that we could crowdfund the developer id :)

**Comment by DanVanAtta** [comment]
The mac users readme file is not really visible until after installation. So a readme file telling you how to install, that is not available until *after* you install, is not very useful.  So we should delete the file and move its contents to somewhere close to the installer (iE: a how to install page somewhere on teh internet that links to the artifacts - (github.io))

It also turns out there is a second easier way to add a security exception. Basically try to open the file twice effectively. It is arguably better method since you don't have to lower a global security setting:https://support.apple.com/kb/PH14369?locale=en_US

Okay, with the game launched, an error pop-up does come up:

> triplea.engine.version.bin:1.8.0.7Could not find root folder, does  not  exist:file:/Applications/TripleA.app/Contents/java/app/bin

> Could not find root folder, does  not exist:file:/Applications/TripleA.app/Contents/java/app/bin


Game can still be played though, so it looks like it is a bug in bad error handling (ie: log an error even though there actually is none).

Otherwise MAC DMG looks fine other than the installer notes not being in the right spot and the non-impactful error message on launch.

**Comment by gaborbernat** [comment]
good question where to add it, maybe we should add it near the download, can't make it part of the installer as that cannot start without that requirement. 

that faulty logging came with the new resource pathing (revert due to the Mac Icon fix) it's a little shame to make that revert as with the new installer the mac icon works fine (as it's no longer part of the game, it's part of the installer). Will try to look into it tomorrow, won't have time sadly until then :worried:

**Comment by gaborbernat** [comment]
on long run I think we should deprecate the all platform zip file, which would allow to impose our own Java version (Mac, Windows bundled into the installer; Linux can be enforced as requirement). this would save some resource as there would be a single Java version to test it against.

**Comment by DanVanAtta** [comment]
Interesting, I'm surprised you didn't suggest we keep only the all_platforms in favor of the OS specific installers. One package that can be tested anywhere and that is the same package seems easier to deal with than the OS specific packages.

The Java bundled download I don't like very much, but I think is still a useful alternative installer to have. Most people should have Java already. Forcing people to use a particularly version of Java I think can invite trouble. There perhaps is a also quick question that we don't violate our own software license by bundling the proprietary oracle JDK with the app.

**Comment by DanVanAtta** [comment]
@gaborbernat , if we get rid of the all_platforms zip, can you describe a bit more how the installation would go on linux?

**Comment by gaborbernat** [comment]
You're wrong, first we bundle just a JRE not a JDK; as that's all the end
users needs. Second we do not enforce our Java version for any application,
just our program (from my understanding). It gets installed right alongside
somewhere tripleA and is used by default just for tripleA.

Keeping just the all_platforms is not realistic. End users don't know how
to install Java, how and where to install their app, and how to uninstall
tripleA/update the package; or to create icons, executebles specific to
their platform. All of this is automatically handled by the installer, with
some configuring. So as you see the installers offer a lot of extra,
platform specific stuff, that we do not need to test (the install4j teams
do that for us). Note that with the installer we no longer pack the run-*,
as those are generated at install time by the installer.

As for license issues see:

Can I distribute Java with my software?

Yes, you can provide Java with your software provided you abide by the
terms and conditions of Java binary code license
<https://www.java.com/en/download/faq/distribution.xml#eula>.

Which essentially states that you can bundle JRE as long as you do it for
running Java apps, as we do.

On my link there is Linux installation tool too. You can try it out. The
idea is the same as on Mac, Linux. A simple choose where, next, next,
finish operation. For the console savy it can even be run from the console
by adding the -q flag (this works on Mac, Windows too from my
understanding, however haven't tried it).

Bernát GÁBOR

**Comment by gaborbernat** [comment]
PS. I actually prefer using the installer on Linux too as it generates a nice menu integrated launcher (with the correct icon) for the app, plus on update asks no more for the path where to install it, just a single next and finish button push is required, and updates everything.

**Comment by DanVanAtta** [comment]
I don't mind being wrong, just curious what I'm wrong about? ;)  Was it that most people already have Java?  Or am I confused on how the Java 6 requirement works?

Regardless, apologies for the confusion regarding the all platform zip, I wasn't intending to seriously propose we keep only the all_platforms zip file. To me it would make more sense to keep that than compared to the other OS specific artifacts. Though, that is a point of formed when we did everything the hard way manually in ant, before we had install4j or install builder in the pipeline.

I'll give the linux installer a try and will let you know how it goes.

**Comment by gaborbernat** [comment]
No, it was the JDK vs JRE part. Plus the licensing issue, plus
enforcing the Java we use on the user. We enforce it just for tripleA,
for none of the other applications.

On 8/30/15, Dan Van Atta <notifications@github.com> wrote:
> I don't mind being wrong, just curious what I'm wrong about? ;)  Was it that
> most people already have Java?  Or am I confused on how the Java 6
> requirement works?
>
> Regardless, apologies for the confusion regarding the all platform zip, I
> wasn't intending to seriously propose we keep only the all_platforms zip
> file. To me it would make more sense to keep that than compared to the other
> OS specific artifacts. Though, that is a point of formed when we did
> everything the hard way manually in ant, before we had install4j or install
> builder in the pipeline.
>
> I'll give the linux installer a try and will let you know how it goes.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/triplea-game/triplea/issues/128#issuecomment-136180067


-- 
 Bernát Gábor
Student -  Budapest University of Technology and Economics - Computer
Engineering, M.Sc. - Budapest, Hungary
System Integrator - Gravity R&D | Rock Solid Recommendations - Budapest
office | 5-7 Expo ter | H-1101 | Budapest | Hungary

**Comment by DanVanAtta** [comment]
Probably good to always be clear about JDK vs JRE. Most people will still have a JRE or a JDK installed.

Licensing issue was a question ;)

> "plus enforcing the Java we use on the user. We enforce it just for tripleA, for none of the other applications."

I don't fully understand this. What are the other applications? Are we enforcing Java, or a specific version of Java?

**Comment by gaborbernat** [comment]
Essentially we are installing the Java bundled just for tripleA, and
using that just for tripleA (not affecting any other applications).

On 8/30/15, Dan Van Atta <notifications@github.com> wrote:
> Probably good to always be clear about JDK vs JRE. Most people will still
> have a JRE or a JDK installed.
>
> Licensing issue was a question ;)
>
>> "plus enforcing the Java we use on the user. We enforce it just for
>> tripleA, for none of the other applications."
>
> I don't fully understand this. What are the other applications? Are we
> enforcing Java, or a specific version of Java?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/triplea-game/triplea/issues/128#issuecomment-136180356


-- 
 Bernát Gábor
Student -  Budapest University of Technology and Economics - Computer
Engineering, M.Sc. - Budapest, Hungary
System Integrator - Gravity R&D | Rock Solid Recommendations - Budapest
office | 5-7 Expo ter | H-1101 | Budapest | Hungary

**Comment by DanVanAtta** [comment]
w.r.t bundled Java. Initially the idea of forcing a Java on someone is not terribly appealing to me. Since our DL size is already 300+MB, an extra 10 or 15MB for bundled Java is a poor consideration. Testing against a known Java version does have appeal. 

So my current thoughts and opinion is: drop the non-bundled Java artifact so we can have fewer artifacts, half as many. It would also help with the test burden. Couple good benefits with currently no known real cost.

 If we find that it truly is not seamless to use the bundled Java, or if we get complaints, then that would be a good time to bring backthe non-bundled version. Alternatively, if we can get our download size to be smaller, 40MB or 50MB, then 10MB would be significant again and we could consider bring back the non-bundled version then.


Also, I just tried out the linux installer. We'll want to be sure we make the installer executable in our build script  before it is pushed. Otherwise it worked really well. With an installer for each OS, I could get on board with removing the all_platforms zip. That would make our build simpler, be one less thing to test, and with an installer for each OS, nobody should be using it.

The linux installer also had the same error on start up message that I saw on mac. I've already lost access to a Mac machine for test, but happily with a repro case on linux for the same issue,  I can most likely it today.

**Comment by ron-murhammer** [comment]
Ant build is complete and artifacts are uploaded to the staging area: https://sourceforge.net/projects/tripleamaps/files/TripleA/stable/1_8_0_9/

I'll see if I can add them to github as well.

**Comment by DanVanAtta** [comment]
@ron-murhammer  -- To upload to github you'll need to tag current master as 1.8.0.9 (should be done anyways, an intentional alignment of process), then you'll see the new version in "releases". Click the edit button next to the release, from there you can drag and drop or select the files.

**Comment by ron-murhammer** [comment]
Yep. Should already be tagged. Issue seems to be with uploading the files. I tried uploading them all at once and it seemed to stop after the first one. Might have to do with the file size. I'm now trying to upload them one at a time but that takes a while...

**Comment by DanVanAtta** [comment]
All_platform zip tested on linux:

- I can't join the lobby with 1.8.0.9, a flaw in our manual regression test plan. Spinning up a test lobby is a good bit of effort, I think a disservice we do to ourselves to make testing harder than it needs to be. Last, if the game engines could play one another seemlessly, it would be easy to get some volunteers to play many *full* games with new engine versions. This play testing would be far more comprehensive than us rushing out test cases.

- map download does work. Though clicking "list games" also opened up a browser window of how to download games along with the DL games dialog window. I was not expecting the browser window, only a minor problem.

Everything else looks good:
- PBEM looks to work, I get dice rolls from Marti.
- single player looks fine, get a round of play in without issue and no naval retreat bug. 
- battle calculator works well

**Comment by ron-murhammer** [comment]
Uploaded to github as well: https://github.com/triplea-game/triplea/releases/tag/1.8.0.9

The map download browser always happens the first time you get a new version (intended feature).

I'll try to do some testing this evening and also get someone to create the mac dmg.

**Comment by DanVanAtta** [comment]
Any  plan for the expected mac DMG bin folder not found error message?

How do you plan to do lobby game testing?

The intended feature does not seem like a feature. I've seen that page already and don't need to see it on every upgrade. In addition, there is another button to do that, and I *did* not click on that. The UI layout should be re-arranged to make the help window button small and off to the corner if we intend to preserve the existing behavior.

**Comment by DanVanAtta** [comment]
@ron-murhammer you can remove the source only zip from github releases. Please note there is a link automatically created for each release with a tar.gz and zip source code only download.

**Comment by DanVanAtta** [comment]
It seems we have decided to release 1.8.0.9 as the next release version (so 1.8.0.7 will not be re-released). We also have decided to treat it as a re-release as 1.8.0.7 for planning purposes (continue Java 6 support, use Ant build, etc).  @ron-murhammer , I encourage a tracking ticket for the 1.8.0.9 release. Seems like current state is regressiong testing of windows installer and all_platforms.zip, and pending request is out for an ANT Mac DMG installer (some good news there, I may have access again to a mac tomorrow  and may be able to build it if macUser doesn't get to it).

Closing since the goal of this issue has been satisified, ie; how to re-release 1.8.0.7
